### PR TITLE
MOB-3180 [UI SDK] Display cancellation SLA in the Booking Request com…

### PR DESCRIPTION
**Pull request by mateid-kr**
MOB-3180

## JIRA
[MOB-3180](https://karhoo.atlassian.net/browse/MOB-3180)

## DESCRIPTION
[UI SDK] Display cancellation SLA in the Booking Request component

## Types of changes

What types of changes does your code introduce to the project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix 
- [X ] New feature 
- [ ] Tech Debt
- [ ] Breaking change 
- [ ] Documentation Update
- [ ] Other: *define what type of change it is*

## CHECKLIST

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. This is simply a reminder of what we are going to look for before merging code._
Please review the guidelines for contributing to this repository.

- [X] All work relating to the ticket is complete (Code complete/Acceptance Criteria met)
- [ ] Lint and unit tests pass locally (if appropriate)
- [X] Added tests that prove the fix is effective or that feature works (if appropriate)
- [ ] Documentation (if appropriate)

## SCREENSHOTS
[Add screenshots if any for changes you have made to the App/UI]

### CODE REVIEW

File: uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/quotes/BookingQuotesView.kt [hunk 1]
```diff
@@ -40,12 +38,18 @@ class BookingQuotesView @JvmOverloads constructor(context: Context,
...
     }
 
-    fun bindViews(url: String?, quoteName: String, category: String) {
+    fun bindViews(url: String?, quoteName: String, category: String, cancellationMinutes: Int?) {
         url?.let { loadImage(it) }
         quoteNameText.text = quoteName
         categoryText.text = String.format("%s%s",
                                           category.substring(0, 1).toUpperCase(),
                                           category.substring(1))
+        if (cancellationMinutes != null && cancellationMinutes > 0) {
```
Comment (jsm20) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/quotes/BookingQuotesView.kt:47]: Can we move this to the presenter so we can unit test the logic?

Comment (jsm20) [inline uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/quotes/BookingQuotesView.kt:47]: Can we move this to the presenter so we can unit test the logic? Seems overkill but I think at some point there might be additional logic required for the View.

Comment (mohadian) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/quotes/BookingQuotesView.kt:47]: We have similar logic here too 
https://github.com/karhoo/karhoo-android-ui-sdk/pull/107/files#diff-421ae67ad42762bca6c3fb0f8a189b0444585c7e1cd94d719c0eb1bde20358cfR179

Perhaps as Jo suggested we can move the logic in Presenter and also share it with other places we need this logic ( very simple condition but worthwhile having one single source of truth)

Comment (mohadian) [reply uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/quotes/BookingQuotesView.kt:47]: the BookingRequestView has reference to context, tou can get the `context.getString(R.string.uisdk_quote_cancellation_minutes)` and then calculate the correct value for `bookingQuoteCancellationText`  ... and then pass the value to 
`bindViews(url: String?, quoteName: String, category: String, cancellationSla: String?)` if the value is null, you can hide the view

**Review by mohadian — APPROVED** [review]

**Pull request merged**